### ISSUE TITLE
SparkTables metadata only delete should throw ValidationException instead of IllegalArgumentException

### ISSUE THREAD

**Issue by szehon-ho** [apache/iceberg#4356]
https://github.com/apache/iceberg/blob/apache-iceberg-0.13.1/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java#L271

As per #4258. seems like IllegalArgumentException is for argument checking only.

**Comment by szehon-ho** [comment]
is it ok to change exception type here?  cc @aokolnychyi

**Comment by Andre9580** [comment]
Hi @szehon-ho, are you suggesting to update the code this way?

```
@Override
  public void deleteWhere(Filter[] filters) {
    Expression deleteExpr = SparkFilters.convert(filters);

    if (deleteExpr == Expressions.alwaysFalse()) {
      LOG.info("Skipping the delete operation as the condition is always false");
      return;
    }

    try {
      icebergTable.newDelete()
          .set("spark.app.id", sparkSession().sparkContext().applicationId())
          .deleteFromRowFilter(deleteExpr)
          .commit();
    } catch (ValidationException e) {
      throw new ValidationException(e, "Failed to cleanly delete data files matching: %s", deleteExpr);
    }
  }
```

**Comment by szehon-ho** [comment]
Hi @Andre9580 yea , even more simply, I think we can even entirely removing the try/catch block

**Comment by Andre9580** [comment]
Hi @[szehon-ho](https://github.com/szehon-ho), created a pull request ([#4630](https://github.com/apache/iceberg/pull/4630)) please feel free to review it.

**Comment by szehon-ho** [comment]
Closed via #4630
